jsonschema-generator/src/test/java/com/github/victools/jsonschema/generator/SchemaGeneratorComplexTypesTest.java

```diff
@@ -216,15 +217,16 @@ public List<? extends TestClass2<Long>> getNestedLongList() {
             return this.nestedLongList;
         }
 
-        public TestClass4<Integer, String> getClass4() {
+        public TestClass4<Integer, String, Long> getClass4() {
             return this.class4;
         }
     }
 
-    private static class TestClass4<S, T> {
+    private static class TestClass4<S, T, P> {
 
         private TestClass2<TestClass2<T>> class2OfClass2OfT;
         public Optional<S> optionalS;
+        public Supplier<P> supplierP;
```

Comment:
Could also just have made this `supplierS` without adding yet another generic. 😉

Reply:
I wasn't sure what to do to address your comment, from #255 turns out it is about not having to use yet another argument `P` but instead reuse the existing `S` leading to:
```java
public Supplier<S> supplierS;
```

Thx!